sql/snapshot.sql

```diff
@@ -0,0 +1,707 @@
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	CONSTRAINT dag_allow_spend_blocks_pkey PRIMARY KEY (round_id),
+	CONSTRAINT dag_allow_spend_blocks_global_snapshot_fk FOREIGN KEY (global_snapshot_hash) REFERENCES global_snapshots(hash)
+);
+
+
+-- public.dag_allow_spends definition
+
+-- Drop table
+
+-- DROP TABLE dag_allow_spends;
+
+CREATE TABLE dag_allow_spends (
+	source_addr varchar NOT NULL,
+	fee int8 NOT NULL,
+	parent_ordinal int8 NULL,
+	parent_hash varchar NULL,
+	last_valid_epoch_progress int8 NOT NULL,
+	round_id uuid NOT NULL,
+	ordinal int8 NOT NULL,
+	CONSTRAINT dag_allow_spends_ordinal UNIQUE (ordinal),
+	CONSTRAINT dag_allow_spends_pk PRIMARY KEY (hash),
+	CONSTRAINT allow_spends_block_fk FOREIGN KEY (round_id) REFERENCES dag_allow_spend_blocks(round_id),
+	CONSTRAINT parent_hash_fk FOREIGN KEY (parent_hash) REFERENCES dag_allow_spends(hash),
+	CONSTRAINT parent_ordinal_fk FOREIGN KEY (parent_ordinal) REFERENCES dag_allow_spends(ordinal)
+)
+INHERITS (public.abstract_transactions);
+
+-- Table Triggers
+
+create trigger set_updated_at_dag_allow_spends before
+update
+    on
+    public.dag_allow_spends for each row execute function update_updated_at_column();
+
+
+-- public.dag_balance_changes definition
+
+-- Drop table
+
+-- DROP TABLE dag_balance_changes;
+
+CREATE TABLE dag_balance_changes (
+	snapshot_hash varchar NOT NULL,
+	address varchar NOT NULL,
+	balance int8 NOT NULL,
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	snapshot_ordinal int8 NOT NULL,
+	CONSTRAINT dag_balance_change_pk PRIMARY KEY (snapshot_ordinal, address),
+	CONSTRAINT dag_balance_change_address_fk FOREIGN KEY (address) REFERENCES addresses(hash),
+	CONSTRAINT dag_balance_change_global_snapshot_fk FOREIGN KEY (snapshot_hash) REFERENCES global_snapshots(hash)
+);
+
+-- Table Triggers
+
+create trigger set_updated_at_dag_balance_change before
+update
+    on
+    public.dag_balance_changes for each row execute function update_updated_at_column();
+
+
+-- public.dag_blocks definition
+
+-- Drop table
+
+-- DROP TABLE dag_blocks;
+
+CREATE TABLE dag_blocks (
+	snapshot_hash varchar NOT NULL,
+	CONSTRAINT dag_block_pk PRIMARY KEY (hash),
+	CONSTRAINT dag_block_global_snapshot_fk FOREIGN KEY (hash) REFERENCES global_snapshots(hash)
+)
+INHERITS (public.abstract_blocks);
+
+-- Table Triggers
+
+create trigger set_updated_at_dag_blocks before
+update
+    on
+    public.dag_blocks for each row execute function update_updated_at_column();
+
+
+-- public.dag_reward_transactions definition
+
+-- Drop table
+
+-- DROP TABLE dag_reward_transactions;
+
+CREATE TABLE dag_reward_transactions (
+	global_snapshot_hash varchar NOT NULL,
+	CONSTRAINT dag_reward_transaction_pk PRIMARY KEY (hash),
+	CONSTRAINT reward_transaction_global_snapshot_fk FOREIGN KEY (global_snapshot_hash) REFERENCES global_snapshots(hash)
+)
+INHERITS (public.abstract_transactions);
```

Comment:
I think it might simplify things if we don't try to have reward transactions inherit from the abstract class. It's an odd one with no source address or hash.